src/lib/voiceSearchHelper/index.ts

```diff
@@ -119,14 +119,17 @@ export default function voiceSearchHelper({
     recognition.start();
   };
 
-  const removeEventListeners = (): void => {
+  const dispose = (): void => {
     if (!recognition) {
       return;
     }
+    recognition.stop();
     recognition.removeEventListener('start', onStart);
     recognition.removeEventListener('error', onError);
     recognition.removeEventListener('result', onResult);
     recognition.removeEventListener('end', onEnd);
+    recognition = undefined;
+    resetState();
```

Comment:
Not sure the `resetState` call is useful, the Helper is destroyed with the widget anyway.

Reply:
I wasn't sure about this. I've removed it now and changed the way to test it.
https://github.com/algolia/instantsearch.js/pull/3779/commits/ae1f20e93ceffacbd4e67be83b6baaefae858f24